NoiseMixture tests in qsimcirq_tests.py fail with Cirq 1.6.0

### Describe the issue

The CI tests fail against Cirq 1.6.0 and the prerelease versions of Cirq. Example run: https://github.com/quantumlib/qsim/actions/runs/16846849990/job/47727587531

The error concerns the `NoiseMixture` parametrized test case in `qsimcirq_tests/qsimcirq_test.py::test_multi_qubit_noise`. Something in this case triggers an exception in `cirq.unitary(…)`:

```
        if isinstance(val, np.ndarray):
            if not linalg.is_unitary(val):
>               raise ValueError("The provided numpy array is not unitary.")
E               ValueError: The provided numpy array is not unitary.

../../../../.pyenv/versions/3.12.5/envs/qsim-py312/lib/python3.12/site-packages/cirq/protocols/unitary_protocol.py:119: ValueError
```

This error does not occur with Cirq 1.5.0.

### What version of qsim are you using?

0.23.0.dev0

### How can the issue be reproduced?

After installing `cirq-core` version 1.6.0 locally, the error can be reproduced by running `pytest`:

```shell
pytest qsimcirq_tests/qsimcirq_tests.py -k test_multi_qubit_noise
```

Doing nothing else but uninstalling `cirq-core` and then installing `cirq-core` version 1.5.0, and repeating the test, results in the tests passing.

--- a/qsimcirq_tests/qsimcirq_test.py
+++ b/qsimcirq_tests/qsimcirq_test.py
@@ -1057,7 +1057,7 @@ class NoiseMixture(NoiseChannel):
         super().__init__(*args, **kwargs)
 
     def _mixture_(self):
-        return [(prob, cirq.unitary(op)) for prob, op, in self._prob_op_pairs]
+        return [(prob, op) for prob, op, in self._prob_op_pairs]
 
 
 @pytest.mark.parametrize(

--- a/qsimcirq_tests/qsimcirq_test.py
+++ b/qsimcirq_tests/qsimcirq_test.py
@@ -2025,11 +2025,11 @@ def test_qsimcirq_identity_expectation_value():
     for w, pauli in objs:
         pauli = pauli[::-1]
         hamiltonian += float(w) * cirq.PauliString(
-            cirq.I(cirq.LineQubit(i))
-            if p == "I"
-            else cirq.Z(cirq.LineQubit(i))
-            if p == "Z"
-            else None
+            (
+                cirq.I(cirq.LineQubit(i))
+                if p == "I"
+                else cirq.Z(cirq.LineQubit(i)) if p == "Z" else None
+            )
             for i, p in enumerate(pauli)
         )
 

--- a/qsimcirq_tests/qsimcirq_test.py
+++ b/qsimcirq_tests/qsimcirq_test.py
@@ -1057,6 +1057,16 @@ class NoiseMixture(NoiseChannel):
         super().__init__(*args, **kwargs)
 
     def _mixture_(self):
+        # Cirq's mixture() function in mixture_protocol.py returns tuples of
+        # the form (probability, unitary peration). It does this by applying
+        # Cirq's unitary() function to the second elements of the tuples
+        # returned from here. Now, the values in self._prob_op_pairs will be
+        # tuples of the form (probability, NoiseStep). NoiseStep defines a
+        # _unitary_() method that simply returns the array as-is. Thus, when
+        # Cirq's mixture() function gets the value returned here and calls
+        # unitary() on those NoiseStep objects, the values unitary() returns
+        # will not actually be unitary. This is done knowingly. The nonunitary
+        # values are eventually normalized in test_multi_qubit_noise().
         return [(prob, op) for prob, op, in self._prob_op_pairs]
 
 

--- a/qsimcirq_tests/qsimcirq_test.py
+++ b/qsimcirq_tests/qsimcirq_test.py
@@ -1058,7 +1058,7 @@ class NoiseMixture(NoiseChannel):
 
     def _mixture_(self):
         # Cirq's mixture() function in mixture_protocol.py returns tuples of
-        # the form (probability, unitary peration). It does this by applying
+        # the form (probability, unitary operation). It does this by applying
         # Cirq's unitary() function to the second elements of the tuples
         # returned from here. Now, the values in self._prob_op_pairs will be
         # tuples of the form (probability, NoiseStep). NoiseStep defines a

